### PR TITLE
doc: Add note about microcode

### PR DESCRIPTION
**Pull request by melwee01**
https://nordicsemi.atlassian.net/browse/NCSDK-30162
https://nordicsemi.atlassian.net/browse/NCSDK-30163

Remove mention of SW crypto from 54L intro
Small cleanup of cryptography.rst page
Add anchor for Configuration header
Add note about CRACEN microcode
Add Kconfig option for microcode download

### CODE REVIEW

File: doc/nrf/app_dev/device_guides/nrf54l/cryptography.rst [hunk 1]
```diff
@@ -24,9 +24,17 @@ KMU and CRACEN peripherals
 **************************
 
 The nRF54L Series Crypto Accelerator Engine (CRACEN) and the Key Management Unit (KMU) peripherals, along with the CRACEN PSA driver, are central when ensuring that the assets of an nRF54L device are protected.
-While CRACEN is not accessed by the CPU and typically not directly used by the end-users and their applications, the KMU provides operations to import, use, revoke, and/or delete assets.
+While CRACEN is not accessed by the CPU and typically not directly used by the end-users and their applications, the KMU provides operations to import, use, revoke, or delete assets.
 Only the KMU is able to push assets to CRACEN's protected RAM and the SEED register.
 
+.. note::
+   CRACEN relies on microcode for asymmetric cryptography operations like signature validation.
+   On nRF54L15, nRF54L10, and nRF54L05 devices, this microcode needs to be uploaded to a special CRACEN RAM area before first use and after reset.
```
Comment (melwee01) [inline doc/nrf/app_dev/device_guides/nrf54l/cryptography.rst:32]: This is listed this way instead of as the Series for futureproofing.

**Review by annwoj — COMMENTED** [review]
very minor suggestions

File: doc/nrf/app_dev/device_guides/nrf54l/cryptography.rst [hunk 2]
```diff
@@ -24,9 +24,17 @@ KMU and CRACEN peripherals
...
+
+   If a bootloader uploads this microcode, there is no need to re-upload it for application use.
```
[the remark this answers is not in the document]
Comment (melwee01) [reply doc/nrf/app_dev/device_guides/nrf54l/cryptography.rst:34]: If I understood correctly, it's done automatically according to a kconfig option, right? In that case, I'm okay with the more generic wording without "you".

[the remark this answers is not in the document]
Comment (tomi-font) [reply doc/nrf/app_dev/device_guides/nrf54l/cryptography.rst:34]: Correct.

File: doc/nrf/app_dev/device_guides/nrf54l/cryptography.rst [hunk 3]
```diff
@@ -303,19 +309,25 @@ You might encounter the following KMU-specific error codes when using the KMU ke
...
 See :ref:`configuring_kconfig` for information on how to set the required configuration options temporarily or permanently.
 
-The CRACEN peripheral does not require any configuration.
-Its operation is ensured by hardware.
+The CRACEN peripheral's operation is ensured by hardware.
+
+The following Kconfig option toggles CRACEN microcode upload:
+
+* :kconfig:option:`CONFIG_CRACEN_LOAD_MICROCODE`: Enabled by default.
+  Set to ``n`` to disable microcode upload.
```
[the remark this answers is not in the document]
Comment (melwee01) [reply doc/nrf/app_dev/device_guides/nrf54l/cryptography.rst:324]: Wouldn't the only reason to disable it be that the user doesn't want CRACEN? If our provided samples don't disable it, what's the scenario where the user would encounter it?

[the remark this answers is not in the document]
Comment (tomi-font) [reply doc/nrf/app_dev/device_guides/nrf54l/cryptography.rst:324]: Yeah but in that case I think it should be a higher-level Kconfig option that manages this, not directly the microcode upload one. (And actually maybe just part of CRACEN would be unusable in that case.) ping @endre-nordic

[the remark this answers is not in the document]
Comment (endre-nordic) [reply doc/nrf/app_dev/device_guides/nrf54l/cryptography.rst:324]: This PR is to document functionality we have right now that enables the application can disable microcode upload if the bootloader already has uploaded it, thus saving flash in the application code.
This needs to be documented for the release.

**Review by endre-nordic — APPROVED** [review]

**Review by annwoj — APPROVED** [review]

**Pull request merged**